### PR TITLE
Prototyping support for running workflows via Kubernetes

### PR DESCRIPTION
**Pull request by scottyw**
This commit adds support for the Lyra Kubernetes controller.

Users can create a custom resource specifying a workflow name, some Hiera data and a refresh time. The Kubernetes controller will run the chosen workflow with the specified frequency.

### CODE REVIEW

File: pkg/apply/apply.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+type Applicator struct {
+	HomeDir string
+}
+
+// ApplyWorkflowWithHieraData will apply the named workflow with the supplied hiera data
+func (a *Applicator) ApplyWorkflowWithHieraData(workflowName string, hieraData map[string]string) {
+	logger := logger.Get()
+
+	// Write the hiera data to a temp file
+	f, err := ioutil.TempFile("", "data-*.yaml")
```
Comment (thallgren) [inline pkg/apply/apply.go]: I think we should avoid taking a detour with a file and YAML for the hiera data. We may just as well just give Hiera a provider function that reads the `hieraData` map directly.

Sample code:
```
func ExampleLookup_mapProvider() {
	sampleData := map[string]string {
		`a`: `value of a`,
		`b`: `value of b`}

	tp := func(ic lookup.ProviderContext, key string, _ map[string]eval.Value) (eval.Value, bool) {
		v, ok := sampleData[key]
		return types.WrapString(v), ok
	}

	lookup.DoWithParent(context.Background(), tp, nil, func(c eval.Context) {
		fmt.Println(lookup.Lookup(impl.NewInvocation(c), `a`, nil, nil))
		fmt.Println(lookup.Lookup(impl.NewInvocation(c), `b`, nil, nil))
	})

	// Output:
	// value of a
	// value of b
}
```

Comment (scottyw) [reply pkg/apply/apply.go]: Nice

**Comment by scottyw** [comment]
I pinned Travis to GO 1.11.1 as a workaround for this: https://github.com/kubernetes/kubernetes/issues/69040

**Comment by kenazk** [comment]
This is so exciting!

File: .travis.yml [hunk 2]
```diff
@@ -7,6 +7,7 @@ jobs:
   include:
   - stage: Test and build
     script:
+    - rm go.sum # Workaround for https://github.com/kubernetes/kubernetes/issues/69040
```
Comment (scottyw) [inline .travis.yml]: Travis is finally green but this ugliness was the only way I could see to achieve it.

@markfuller @thallgren What do you reckon?

Comment (markfuller) [reply .travis.yml]: I think this is the best option at this point.

**Comment by kenazk** [comment]
Supports #37

**Pull request merged**